src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleFileFunction.java

```diff
@@ -529,7 +533,8 @@ private static ModuleThreadContext execModuleFile(
    *
    * @param registry can be null if this module has a non-registry override.
    */
-  private record GetModuleFileResult(ModuleFile moduleFile, @Nullable Registry registry) {}
+  private record GetModuleFileResult(
+      ModuleFile moduleFile, @Nullable Registry registry, StoredEventHandler downloadEvents) {}
```

Comment:
s/downloadEvents/downloadEventHandler/ ? reason I'm suggesting is that calling it `downloadEvents` may make it seem immutable, and is slightly confusing for call sites like line 624.